cmake/rename.cmake

```diff
@@ -13,8 +13,9 @@ macro(gko_rename_cache deprecated actual type doc_string)
         else()
             # Only set `deprecated`, move it to `actual`.
             message(WARNING "${deprecated} was deprecated, please use ${actual} instead.  "
-                "We copy ${${deprecated}} to ${actual}")
+                "We copy ${${deprecated}} to ${actual} and remove the old variable")
             set(${actual} ${${deprecated}} CACHE ${type} "${doc_string}")
+            unset(${${deprecated}} CACHE)
```

Comment:
should this be outside the `if(DEFINED ${actual})`? After both branches, there is still the deprecated variable and the new one.

Reply:
That's probably okay to do, I just didn't want to touch cases where the function was already run previously.

Reply:
I added `unset` in to places instead, to make the control-flow involving `FATAL_ERROR` more clear